clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -271,16 +271,22 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
   } else {
     // Output directory is based off of the first object name
     for (Arg * Cur : Args) {
-      SmallString<128> AN = Cur->getSpelling();
-      StringRef Ext(llvm::sys::path::extension(AN));
-      if (!Ext.empty()) {
-        types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
-        if (Ty == types::TY_INVALID)
-          continue;
-        if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
-          llvm::sys::path::replace_extension(AN, "prj");
-          ReportOptArg += Args.MakeArgString(AN);
-          break;
+      if (Cur->getOption().getKind() == Option::InputClass) {
```

Comment:
Is an early exit possible here?

Reply:
naming is based on source or object, so we can't guarantee early exit just based on first input seen. (if that is what you were getting at)

Reply:
Oh, sorry for the confusion: I've rather meant an early `continue` in iterating over `Args`

Reply:
Given that there is nothing else to skip over if this condition isn't met I don't see any opportunity for an early exit.  I'm open to suggestions if you see an opportunity.